demos/TicTacToe/Classes/TicTacToeScene.cpp

```diff
@@ -0,0 +1,78 @@
+#include "TicTacToeScene.h"
+
+#include <array>
+#include <cstdlib>
+
+#include "cocos2d.h"
+
+USING_NS_CC;
+
+static const int kTilesX = 3;
+static const int kTilesY = 3;
+static const int kNumberOfTiles = kTilesX * kTilesY;
+static const int kMaxMovesPerPlayer = 1 + kNumberOfTiles / 2;
+static const double kScreenWidth = 600;
+static const double kScreenHeight = 600;
+static const double kTileWidth = (kScreenWidth / kTilesX);
+static const double kTileHeight = (kScreenHeight / kTilesY);
+static const int kNumberOfPlayers = 2;
+static const char* kBoardImageFileName = "tic_tac_toe_board.png";
+std::array<const char*, kNumberOfPlayers> kMoveImageFileNames = {
+    "tic_tac_toe_x.png", "tic_tac_toe_o.png"};
+
+Scene* TicTacToe::createScene() {
+  auto scene = Scene::create();
```

Comment:
Since the return parameter is defined as Scene*, my preference would be define the 'scene' variable as Scene* here as well instead of auto.

Reply:
Could any of these return a null pointer?

Reply:
Both autos were swapped to their respective type.

The null pointer question is a good one and I am not sure. I'll look at some of the cocos documentation for createScene() and init() then loop around here